src/py/flwr/server/server_app.py

```diff
@@ -55,32 +57,39 @@ class ServerApp:
 
     def __init__(
         self,
-        server: Optional[Server] = None,
```

Comment:
We should not remove the `Server` argument to enable users to use `ServerApp` with existing `Server` implementations. My perspective is that there are two modes to use `ServerApp`:
- Legacy mode via `Server`/`Strategy`/`ServerConfig`/`ClientManager`
- New mode via `Workflow` and/or `@app.main()`